app/src/main/java/org/mozilla/fenix/downloads/DownloadNotification.kt

```diff
@@ -0,0 +1,143 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.downloads
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.download_notification_layout.view.*
+import mozilla.components.browser.state.state.content.DownloadState
+import mozilla.components.feature.downloads.AbstractFetchDownloadService
+import mozilla.components.feature.downloads.toMegabyteString
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.toolbar.BrowserToolbarView
+import org.mozilla.fenix.ext.metrics
+import org.mozilla.fenix.ext.settings
+
+/**
+ * [DownloadNotification] is used to show a view in the current tab to the user, triggered when
+ *  downloadFeature.onDownloadStopped gets invoked. It uses [DownloadNotificationBehavior] to
+ *  hide when the users scrolls through a website as to not impede his activities. Uses
+ *  [DownloadState.dismissed] to track whether the user has closed the view.
+ * */
+
+class DownloadNotification(
```

Comment:
As mentioned on https://github.com/mozilla-mobile/android-components/pull/6899#discussion_r425428064, I think it will be better to rename this class to include the word dialog as `DownloadNotification` can be a source of confusion with the download notifications in the status bar.